### DISCUSSION TITLE
Precargado de múltiples propiedades en Question Exec

### DISCUSSION THREAD

**Discussion by maramire** [Cotalker/documentation#855] [2. Questions and Answers] [answered]
Holaa, buenas

Estoy intentando precargar propiedades en un multiple choice, pero no me deja precargar más de una propiedad. Este es mi código del question Exec (donde transacciones es un array de id's de propiedades):

function run(){
    const commands = [];
    const property = context['property#channel'].find(k => k.propertyType === "controversias_assets");
    const transacciones = property.schemaInstance.transacciones;
    console.log(JSON.stringify(transacciones))
    commands.push(
        {
        cmd: 'SET_RESPONSES',
        value: JSON.stringify(transacciones)
        }
    );
    commands.push(
      { 
      cmd: 'SET_READONLY',
      value: 'true' 
      }
    );
    return commands;
}

Esto es lo que me retorna la llamada de red:
![image](https://user-images.githubusercontent.com/31930273/164057778-a12ba170-cece-43a5-bb03-693d93cc177f.png)

Probé precargando solo una propiedad y funcionaba, pero al mandar un array de id's me devuelve el error antes mencionado.

Se agradece cualquier ayuda!

**Comment by guillevalin** [comment]
@acedward @agallardol a mi también me ocurrió algo raro, bajo el modelo de datos de un answer y de una question de tipo survey+property, el "responses" es un array de properties en formato JSON String. Sin embargo, aquí el preload pasando un array de properties me da error justamente en la llamada de red que muestra @maramire porque intenta pasar el objeto completo.

Lo que me funciona es lo siguiente:
function run(){
return [{ cmd: "SET_RESPONSES", value: "5e38aaff0bfe796f37443d66"}];
}

Resultado:
![image](https://user-images.githubusercontent.com/34345309/164071695-111db34b-c185-4a01-bbce-0be13ece06f5.png)

Sin embargo, si hago lo siguiente:
function run(){
return [{ cmd: "SET_RESPONSES", value: ["5e38aaff0bfe796f37443d66"]}];
}

La query que muestra Matías sigue funcionando igual y retorna 200, pero no se precarga la question:
https://www.cotalker.com/api/v2/properties?ids=5e38aaff0bfe796f37443d66&limit=100&page=1&count=true

Y si hago lo siguiente:
function run(){
return [{ cmd: "SET_RESPONSES", value: ["5e38aaff0bfe796f37443d66", "5e38ab074f29c06eb87bbe99"]}];
}

Se rompe el formulario:
![image](https://user-images.githubusercontent.com/34345309/164074634-8aa7aed2-1c62-44b4-b653-b1dbf082fe27.png)

Trata de hacer una query de red por las properties, pero da BAD_REQUEST:
https://www.cotalker.com/api/v2/properties?ids=5e38aaff0bfe796f37443d66,5e38ab074f29c06eb87bbe99&limit=100&page=1&count=true

**Comment by guillevalin** [comment, marked as answer]
Se levantó el issue #856

**Comment by acedward** [comment]
Voy a marcar como cerrado. Para dejar la información en el issue.
